src/tests/test_definitions.py

```diff
@@ -126,6 +126,36 @@ def nested_3(self):
 from __future__ import (, )
 """
 
+source_invalid_syntax = """
+#: E901
+}
+#: E901
+= [x
+#: E901 E101 W191
+while True:
+    try:
+	    pass
+	except:
+		print 'Whoops'
+#: E122 E225 E251 E251 E701
+
+# Do not crash if code is invalid
+if msg:
+    errmsg = msg % progress.get(cr_dbname))
+
+def lasting(self, duration=300):
+    progress = self._progress.setdefault('foo', {}
+#: Okay
+
+# Issue #119
+# Do not crash with Python3 if the line endswith '\r\r\n'
+EMPTY_SET = set()
+SET_TYPE = type(EMPTY_SET)
+
+toto = 0 + 0
+#:
+"""  # noqa
+
```

Comment:
This snippet is the test file that was reported to cause issues in #168, but yes, we can minimize the test example. Turns out that the single invalid token (e.g. "[" like you mentioned) leads to a different exception: tk.TokenError so I added that as a test case too.